lib/service/service.go

```diff
@@ -4258,6 +4277,7 @@ func (process *TeleportProcess) getAdditionalPrincipals(role types.SystemRole, h
 //  4. optionally proxy kubernetes connections
 //  5. optionally check for automatic upgrades for deployments created by AWS OIDC integrations
 func (process *TeleportProcess) initProxy() error {
+	process.ExpectService(teleport.ComponentProxy)
```

Comment:
I think there there is no heartbeat  mechanism implemented for the proxy process. In the result when the `process.ExpectService(teleport.ComponentProxy)` is emitted the process will never report readiness.

You can verify this by starting teleport with the `proxy_services`  and checking the probe with the changes from this PR: 

```
curl  http://127.0.0.1:3333/readyz
{"status":"teleport is starting and hasn't joined the cluster yet","pid":57186
```